OpenRA.Mods.Common/AI/States/GroundStates.cs

```diff
@@ -130,19 +140,21 @@ public void Tick(Squad owner)
 
 			if (!owner.IsTargetValid)
 			{
-				var closestEnemy = owner.Bot.FindClosestEnemy(owner.Units.Random(owner.Random).CenterPosition);
-				if (closestEnemy != null)
-					owner.TargetActor = closestEnemy;
+				var t = FindClosestEnemy(owner);
+				if (t != null)
+					owner.TargetActor = t;
 				else
 				{
 					owner.FuzzyStateMachine.ChangeState(owner, new GroundUnitsFleeState(), true);
 					return;
 				}
 			}
 
+			// Switch target durign fight
+			var targetActor = owner.Bot.FindClosestEnemy(owner.Units.First().CenterPosition);
 			foreach (var a in owner.Units)
 				if (!BusyAttack(a))
-					owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = owner.Bot.FindClosestEnemy(a.CenterPosition) });
+					owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = targetActor });
```

Comment:
Only units not attacking are given new targets each tick. This will only be faster if multiple units in the squad all become idle in the same tick, and can thus be assigned the same target. Hopefully that holds true. :)

However, moving the target outside the loop means you have to find a new target, even if nobody ends up using it. You should only find a new target when you need it, e.g:

```c#
Actor targetActor = null;
foreach (var a in owner.Units)
{
    if (!BusyAttack(a))
    {
        if (targetActor == null)
            targetActor = owner.Bot.FindClosestEnemy(a.CenterPosition);

		owner.Bot.QueueOrder(new Order("Attack", a, false) { TargetActor = targetActor });
    }
}
```
This means all the units that become idle in the same tick will gang up on the same target as the first idle unit.